### PR TITLE
New blog post - Performance testing and Gatling

### PR DESCRIPTION
**Pull request by nzhang0619**
In a comment on the pull request, include the following information and check off the list.

 - [x] Mini bio
  - Soft capped to 75 words
Ning Zhang is a Software Developer in Test in the Rackspace Private Cloud Powered by VMware. Ning Zhang has years of experience in automation testing, performance testing, agile development and CI/CD.

 - [x] Desired date of publishing
  - 2018-12-15
 - [ ] Social media contact info
 - [x] Redaction: YES
  - Allow redaction, editing or structure changes by technical writer staff
 - [x] Comments: true

@catlook Please review.

### CODE REVIEW

**Comment by rackernexus** [comment]
HULK [PREVIEW](https://staging.developer.rackspace.com/build-88e7339342/)

File: _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+
+## Gatling tool for your performance testing
+
+Gatling is an open-source load and performance testing framework based on Scala, Akka and Netty, and allows you to run it on any system. It is designed for ease of use, maintainability and high performance on different local machines and cloud servers to create and run your tests. 
+
+Out of the box, Gatling generates detailed metrics dashboard of test results, which are stored as HTML files for deep analysis and metrics comparison, including indicators, active users, number of requests, and response time.
+
```
Comment (catlook) [inline _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md]: The following images illustrate each of the metrics in the dashboard:

File: _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+#### Response time percentiles over time:
+
+{% img center 2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool/charts-response-percentiles-per-sec.png %}
+
+#### Response time distribution:
+
+{% img center 2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool/charts-distrib.png %}
+
+In addition, Gatling performance tool has built-in integration with Continuous Integration pipelines. For instance, you can choose the useful [Jenkins Gatling plugin](https://plugins.jenkins.io/gatling), which enables you to keep track of a Gatling simulation providing performance trends across builds, and publish detailed reports for each build.
```
Comment (catlook) [inline _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md]: ... has a built-in  ...
... with *continuous integration* pipelines ...
... builds and to publish  ...

File: _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+
+{% img center 2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool/jenkins-dashboard.png %}
+
+### How to run a simple performance test with Gatling
+
+#### Installing
+Go to Gatling [download page](https://gatling.io/download/), and download the latest Gatling tool zip bundle. Unzip and put it in some location.
```
Comment (catlook) [inline _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md]: ... Gatling [download page](https://gatling.io/download/) and download ...
... Unzip it and store it on your server. ...

File: _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+
+#### Configuring your browser
+Before starting to record a test scenario using Gatling recorder, we need to configure our browser proxy settings with the following steps. (All steps are done with a Google Chrome browser on Mac in this demo.)
+1. Open the Google Chrome browser, and go to the settings.
+2. Open proxy settings, and click on **Advanced** button on the bottom of the page.
```
Comment (catlook) [inline _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md]: ... click **Advanced** at the ...

File: _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+
+    {% img center 2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool/configuring-browser-1.png %}
+
+    {% img center 2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool/configuring-browser-2.png %}
+
+3. Go to Proxies tab, check **Web Proxy (HTTP)** and **Secure Web Proxy (HTTPS)**, and type the address **127.0.0.1** and the port **8000**.
```
Comment (catlook) [inline _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md]: ... **Secure Web Proxy (HTTPS)**. Type  ...

File: _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+
+    {% img center 2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool/configuring-browser-4.png %}
+    
+4. Close the browser window.
+
+#### Recording a test scenario
+
+1. Gatling recorder `recorder.sh` is located under _/gatling-charts-highcharts-bundle-3.0.1.1/bin_. Note that `Java.exe` is required to open the recorder window.
+2. Before starting the recorder, enter some settings in the recorder window.
+* Enter the port number in the local host box (In my case **8000**)
```
Comment (catlook) [inline _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md]: ... box. (I used **8000**). ...

File: _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+* Enter the package name and the class name
```
Comment (catlook) [inline _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md]: ... class name. ...

File: _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+* Check the **Follow Redirects?**, **Infer HTML resources?**, **Remove cache Headers?** and **Automatic Referers?**
```
Comment (catlook) [inline _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md]: Check **Follow Redirects?**, **Infer HTML resources?**, **Remove cache Headers?**, and **Automatic Referers?**.

File: _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+* Select the output folder path
```
Comment (catlook) [inline _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md]: .... folder path. ...

File: _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+* Keep all other options as default
```
Comment (catlook) [inline _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md]: ... as default. ...

File: _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md [hunk 11]
```diff
@@ -0,0 +1,173 @@
...
+* Click on the **Start** button
```
Comment (catlook) [inline _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md]: ...  Click **Start**. ...

File: _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md [hunk 12]
```diff
@@ -0,0 +1,173 @@
...
+
+{% img center 2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool/gatling-recorder-1.png %}
+
+3. Open the Google Chrome browser, go to the URL: http://computer-database.gatling.io/, and try some operations on this page. The executed events can be observed in the recorder window. Click on the **Stop & Save** button.
```
Comment (catlook) [inline _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md]: ... Click **Stop & Save**. ...

File: _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md [hunk 13]
```diff
@@ -0,0 +1,173 @@
...
+#### Executing a Gatling script
+Launch `gatling.sh` located in the bin directory, and you should see a menu with the simulation examples:
+
+{% img center 2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool/gatling-script-output.png %}
+
+And select the id number of the simulation you want to execute. When the simulation is done, the console will display a link to the HTML reports.
+
+### Integrate Gatling with Maven
+
+This section will show you how to integrate Gatling tool with Maven via plugins, which has become essential in the world of CI/CD.
```
Comment (catlook) [inline _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md]: ... This section shows you how  ...

[Editor note: what is CI/CD?]

File: _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md [hunk 14]
```diff
@@ -0,0 +1,173 @@
...
+
+#### Plugin versions and download
+To use Gatling with Maven, the plugin `gatling-maven-plugin` is required. Since Gatling is written in Scala, you also need another plugin `scala-maven-plugin` for it. Check out available versions on [Maven Central](https://search.maven.org/search?q=g:io.gatling%20AND%20a:gatling-maven-plugin&core=gav).
+
+#### Setup and configurations
+After creating a Maven project on your local, the setup in pom.xml file looks more or less like this:
```
Comment (catlook) [inline _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md]: ... your local machine, the ...
... in the **pom.xml** file ...

File: _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md [hunk 15]
```diff
@@ -0,0 +1,173 @@
...
+
+{% img center 2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool/pom-xml.png %}
+
+The configurations can also be done in pom.xml file, like the multiple simulations’ execution, includes/excludes filters, etc. Check the [Gatling documentation](https://gatling.io/docs/current/extensions/maven_plugin/#configuration) for more configuration examples.
```
Comment (catlook) [inline _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md]: In **pom.xml**, you can configure things like multiple simulations’ ....

File: _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md [hunk 16]
```diff
@@ -0,0 +1,173 @@
...
+To execute your tests, you can directly launch the gatling-maven-plugin with the test or execute goal:
+```
+mvn gatling:test
+mvn gatling:execute
+```
+
+
+## Other performance testing tools
+
+Below is a list of most widely used load and performance testing tools, whether those be open source, paid or a combination of both.
```
Comment (catlook) [inline _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md]: The following list has the most widely ...

File: _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md [hunk 17]
```diff
@@ -0,0 +1,173 @@
...
+```
+
+
+## Other performance testing tools
+
+Below is a list of most widely used load and performance testing tools, whether those be open source, paid or a combination of both. 
+
+#### Apache JMeter
+
+Apache JMeter is a Java-based open source tool for load testing and measuring performance. It’s used for recording, building, monitoring and debugging on different applications, servers and networks. Also, it is of a great use in creating a functional or load test plan.
```
Comment (catlook) [inline _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md]: ... monitoring, and debugging ...
... servers, and networks. ...

File: _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md [hunk 18]
```diff
@@ -0,0 +1,173 @@
...
+Download link: [Apache JMeter download](http://jmeter.apache.org/download_jmeter.cgi)
+
+#### WebLOAD
+
+WebLoad is an enterprise-level load and performance testing tool for web applications. It has both a free version and a paid version. The free edition offers 50 virtual users, and the paid professional edition provides the choices for enterprises with heavy user load and complex testing requirements. WebLOAD Cloud has a new level of visibility, control and collaboration – on premises or in the cloud, and it integrates with other open source software like Selenium and Jenkins.
```
Comment (catlook) [inline _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md]: WebLOAD is ...
... control, and collaboration ...

File: _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md [hunk 19]
```diff
@@ -0,0 +1,173 @@
...
+Free trial: [WebLOAD download](https://www.radview.com/webload-download/?utm_campaign=top-15-tools&utm_medium=top-15-tools&utm_source=softwaretestinghelp)
+
+#### LoadNinja
+
+LoadNinjia is a cloud-based load testing and performance testing platform for web applications. It helps developers, QA teams, and performance engineers check if their web servers sustain a massive load and if the servers are robust and scalable, and also helps the engineer to focus more on creating applications that scale instead of focusing on developing load testing scripts.
```
Comment (catlook) [inline _posts/2018-11-29-Introduction-to-load-and-performance-testing-with-gatling-tool.md]: .. scalable. It also helps ...

**Comment by rackernexus** [comment]
:mag_right: [Preview](https://staging.developer.rackspace.com/build-468a2e50c8/)

**Comment by nzhang0619** [comment]
@catlook I made the changes based on your comments, and pushed them up. Please review. Thanks

**Comment by rackernexus** [comment]
:mag_right: [Preview](https://staging.developer.rackspace.com/build-468a2e50c8/)

**Pull request merged**